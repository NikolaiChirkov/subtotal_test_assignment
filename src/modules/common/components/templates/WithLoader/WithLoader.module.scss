.root {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 100vh;
  background-color: black;

  &__text {
    margin-top: 8px;
    font-size: 16px;
    font-weight: 600;
    color: white;
  }
}
