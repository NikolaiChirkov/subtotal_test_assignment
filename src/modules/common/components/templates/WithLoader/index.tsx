import React from "react";
import { Loader } from "../../atoms";
import { FCWithChildren } from "@modules/common/types";
import styles from "./WithLoader.module.scss";

export type WithLoaderProps = {
  isLoading: boolean;
};

export const WithLoader: FCWithChildren<WithLoaderProps> = ({
  isLoading,
  children,
}) => {
  if (isLoading) {
    return (
      <div className={styles.root}>
        <Loader />
        <p className={styles.root__text}>Please wait, data is loading...</p>
      </div>
    );
  }

  return children;
};
