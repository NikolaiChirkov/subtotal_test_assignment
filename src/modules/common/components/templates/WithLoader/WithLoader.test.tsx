import React from "react";
import renderer from "react-test-renderer";
import { WithLoader } from "../WithLoader";

it("component changes corresponding to data loading", () => {
  const component = renderer.create(
    <WithLoader isLoading={true}>test</WithLoader>
  );
  
  let tree = component.toJSON() as renderer.ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.isLoading = false;
  });

  tree = component.toJSON() as renderer.ReactTestRendererJSON;

  renderer.act(() => {
    tree.props.isLoading = true;
  });

  tree = component.toJSON() as renderer.ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});
