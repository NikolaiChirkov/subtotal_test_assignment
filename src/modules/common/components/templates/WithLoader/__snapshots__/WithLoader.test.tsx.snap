// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component changes corresponding to data loading 1`] = `
<div>
  <svg
    height="48"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z"
    />
  </svg>
  <p>
    Please wait, data is loading...
  </p>
</div>
`;

exports[`component changes corresponding to data loading 2`] = `
<div>
  <svg
    height="48"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10.14,1.16a11,11,0,0,0-9,8.92A1.59,1.59,0,0,0,2.46,12,1.52,1.52,0,0,0,4.11,10.7a8,8,0,0,1,6.66-6.61A1.42,1.42,0,0,0,12,2.69h0A1.57,1.57,0,0,0,10.14,1.16Z"
    />
  </svg>
  <p>
    Please wait, data is loading...
  </p>
</div>
`;
